src/harness/os/linux.c

```diff
@@ -285,3 +284,28 @@ void OS_InstallSignalHandler(char* program_name) {
         }
     }
 }
+
+FILE* OS_fopen(const char* pathname, const char* mode) {
+    FILE* f = fopen(pathname, mode);
+    if (f != NULL) {
+        return f;
+    }
+    char buffer[512];
+    char buffer2[512];
+    strcpy(buffer, pathname);
+    strcpy(buffer2, pathname);
+    char* pDirName = dirname(buffer);
+    char* pBaseName = basename(buffer2);
+    DIR* pDir = opendir(pDirName);
```

Comment:
if we re-write this using `OS_FindFirstFile` and `OS_FindNextFile`, is there any reason we couldn't move this functionality into `Harness_Hook_fopen`? If thats possible, that would be nice, since that would be the single place for getting a `FILE *` handle. Loading a file from an archive or straight from an http request etc belong in a non-OS specific place.

for example:
```
FILE* f = fopen(pathname, mode);
if (f != NULL) {
    return f;
}
...
char* file_found = OS_GetFirstFileInDirectory(pathname);
    while (file_found != NULL) {
        if (strcasecmp(pBaseName, file_found) == 0) {
            strcat(pDirName, "/");
            strcat(pDirName, file_found);
            return fopen(pDirName, mode);
       }
       file_found = OS_GetNextFileInDirectory(pathname);
    }
```

Reply:
Also, MacOS _can_ run with a case sensitive filesystem (https://coderwall.com/p/mgi8ja/case-sensitive-git-in-mac-os-x-like-a-pro), so the same code that currently exists here in `linux.c` would otherwise need to duplicated into `macos.c`. Maybe irrelevant but Haiku OS is also case sensitive if we ever want to support it

Reply:
We can also move this case insensitive search to a separate file + header such that different operating systems can make use of it.